exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StorageResources.java

```diff
@@ -276,24 +280,24 @@ public Response updateAccessToken(@PathParam("name") String name, OAuthTokenCont
   public Response updateOAuthTokens(@PathParam("name") String name,
                                     OAuthTokenContainer tokenContainer) {
     try {
-      if (storage.getPlugin(name).getConfig() instanceof CredentialedStoragePluginConfig) {
-        DrillbitContext context = ((AbstractStoragePlugin) storage.getPlugin(name)).getContext();
-        OAuthTokenProvider tokenProvider = context.getoAuthTokenProvider();
-        PersistentTokenTable tokenTable = tokenProvider.getOauthTokenRegistry().getTokenTable(name);
-
-        // Set the access and refresh token
-        tokenTable.setAccessToken(tokenContainer.getAccessToken());
-        tokenTable.setRefreshToken(tokenContainer.getRefreshToken());
-
-        return Response.status(Status.OK)
-          .entity("Access tokens have been updated.")
-          .build();
-      } else {
+      StoragePluginConfig config = storage.getPlugin(name).getConfig();
+      if (config.getValue("type") != "http") {
```

Comment:
The intention was to allow it not only for HTTP but for all other plugins that have `CredentialedStoragePluginConfig`, so they theoretically can also use OAuth and update tokens.

Reply:
But looks like error message below should be fixed

Reply:
@jnturton This is actually addressed in https://github.com/apache/drill/pull/2544.   I refactored the OAuth stuff so that if other plugins needed to use OAuth, they can reuse almost all of what Drill already does.